src/snippet/mod.rs

```diff
@@ -566,4 +580,42 @@ Survey in 2016, 2017, and 2018."#;
         }
         Ok(())
     }
+
+    #[test]
+    fn test_snippet_generator_json_field() -> crate::Result<()> {
+        let mut schema_builder = Schema::builder();
+        let json_object_options = JsonObjectOptions::default().set_indexing_options(
+            TextFieldIndexing::default()
+                .set_tokenizer("en_stem")
+                .set_index_option(IndexRecordOption::WithFreqs),
+        );
+        let json_field = schema_builder.add_json_field("json", json_object_options);
+        let schema = schema_builder.build();
+        let index = Index::create_in_ram(schema);
+
+        let json_val: serde_json::Map<String, serde_json::Value> =
+            serde_json::from_str(r#"{ "test": "this is the design of rust" }"#).unwrap();
+        {
+            // writing the segment
+            let mut index_writer = index.writer_for_tests()?;
+            let doc = doc!(json_field => json_val.clone());
+            index_writer.add_document(doc)?;
+            index_writer.commit()?;
+        }
+        let searcher = index.reader().unwrap().searcher();
+        let query_parser = QueryParser::for_index(&index, vec![json_field]);
+        let query = query_parser
+            .parse_query("json.test:rust json.test:design")
+            .unwrap();
+        let snippet_generator = SnippetGenerator::create(&searcher, &*query, json_field).unwrap();
+        {
+            let doc = doc!(json_field => json_val.clone());
+            let snippet = snippet_generator.snippet_from_doc(&doc);
+            assert_eq!(
+                snippet.to_html(),
+                "this is the <b>design</b> of <b>rust</b>"
```

Comment:
Does this method should generate `{"test": "this is the <b>design</b> of <b>rust</b>"}`?

Reply:
Ah I see..!

Reply:
@lavrd @fulmicoton 
Hi, sorry for the late response.  When the nest is deep, should it return
```
{"a": {"b": {"c": "this is the <b>design</b> of <b>rust</b>"}}}
```

Or 
```
{"c": "this is the <b>design</b> of <b>rust</b>"}
```

Reply:
Hi, I think it is more a question for maintainers, but I expect and want to have first variant with whole json. Also, if you can implement both variants, maybe you can add an option to chose deep level? Because maybe other users want to have another deep level 😅